### PR TITLE
Non virtual dtor

### PR DESCRIPTION
**Pull request by janisozaur**
part of https://github.com/OpenRCT2/OpenRCT2/issues/4401

### CODE REVIEW

File: src/core/StringReader.hpp [hunk 1]
```diff
@@ -27,6 +27,7 @@ interface IStringReader
     virtual bool TryRead(codepoint_t * outCodepoint)       abstract;
     virtual void Skip()                                    abstract;
     virtual bool CanRead()                           const abstract;
+    virtual ~IStringReader() = default;
```
Comment (IntelOrca) [inline src/core/StringReader.hpp]: Move to top and put a line break after it like I did for other interfaces.
What does default mean again? Is it same as `{ }`?

Comment (janisozaur) [inline src/core/StringReader.hpp]: pretty much. http://en.cppreference.com/w/cpp/language/default_constructor

Comment (IntelOrca) [inline src/core/StringReader.hpp]: ah ok, feel free to change all the other places I have done braces as well.

**Comment by janisozaur** [comment]
@IntelOrca updated

**Pull request merged**